samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -11,18 +11,35 @@
 
 #define GET_DATA_INTERVAL 60000
 #define GET_DATA_INTERVAL_QUICK 3000
+#define MOTION_TIMEOUT K_SECONDS(60)
+
+static bool is_occupied;
+static struct k_work_delayable motion_timeout_work;
+
+static void motion_timeout(struct k_work *work)
+{
+	is_occupied = false;
+	printk("Area is now vacant.\n");
+}
 
 static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
 			       struct bt_mesh_msg_ctx *ctx,
 			       const struct bt_mesh_sensor_type *sensor,
 			       const struct bt_mesh_sensor_value *value)
 {
+	enum bt_mesh_sensor_value_status status;
+
 	if (sensor->id == bt_mesh_sensor_present_dev_op_temp.id) {
 		printk("Chip temperature: %s\n", bt_mesh_sensor_ch_str(value));
 	} else if (sensor->id == bt_mesh_sensor_presence_detected.id) {
 		int64_t presence_detected = 0;
 
-		(void)bt_mesh_sensor_value_to_micro(value, &presence_detected);
+		status = bt_mesh_sensor_value_to_micro(value, &presence_detected);
+		if (status != BT_MESH_SENSOR_VALUE_NUMBER) {
+			printk("Warning: unexpected behaviour during conversion of presence "
```

Comment:
If the code ends up here, we have no guarantees at all about the contents of `presence_detected`, and the code should stop processing and not continue like it currently does. Same for the Motion Sensed-sensor.

Reply:
I'd also prefer to use `bt_mesh_sensor_value_status_is_numeric` instead of comparing to `BT_MESH_SENSOR_VALUE_NUMBER` directly, in all places where this is done. It will not make a difference for these specific sensor types (which don't support less than/greater than), but the best practice to check "did I get a number out of this" is to check for `...is_numeric`.